poc/flp_pine.py

```diff
@@ -269,12 +252,18 @@ def eval_norm_equality_check(self,
         """Check that the computed squared L2-norm result matches the value
         claimed by the Client. """
         # Compute the squared norm.
-        mul_inputs = []
-        for val in encoded_gradient:
-            mul_inputs += [val, val]
-        computed_sq_norm = self.parallel_sum(mul_inputs)
-        sq_norm_v = self.field.decode_from_bit_vector(sq_norm_v_bits)
+        computed_sq_norm = self.field(0)
+        while len(encoded_gradient) >= self.chunk_length:
```

Comment:
I think we can combine the functionality into `parallel_sum`, parameterized by `gadget.arity`, so we can put the right amount of inputs into the chunk? Something like:

```
        g = self.GADGETS[0]
        s = self.field(0)
        while len(inputs) >=  g.ARITY * self.chunk_length:
            (chunk, inputs) = front(g.ARITY * self.chunk_length, inputs)
            s += g.eval(self.field, chunk)
        if len(inputs) > 0:
            chunk = self.field.zeros(g.ARITY * self.chunk_length)
            for i in range(len(inputs)):
                chunk[i] = mul_inputs[i]
            s += g.eval(self.field, chunk)
        return s
```

where `inputs` is `mul_inputs` for `Mul` gadget, and is `encoded_gradient` for `PolyEval` gadget.

Reply:
Good idea, done.